scripts/compile_sass.py

```diff
@@ -153,42 +155,108 @@ def main(
                 return
             else:
-                raise FileNotFoundError(f"missing Sass source dir: {source}")
-        click.echo(f"      Include paths:")
+                raise FileNotFoundError(f"missing Sass source dir: {source_root}")
+        click.echo(f"    Include paths:")
         for include in includes:
-            click.echo(f"        {include}")
-        if not dry:
-            # Import sass late so that this script can be dry-run without installing
-            # libsass, which takes a while as it must be compiled from its C source.
-            import sass
-
-            dest.mkdir(parents=True, exist_ok=True)
-            sass.compile(
-                dirname=(str(source), str(dest)),
-                include_paths=[str(include_path) for include_path in includes],
-                source_comments=use_dev_settings,
-                output_style=("nested" if use_dev_settings else "compressed"),
-            )
-        click.secho(f"      Compiled.", fg="green")
+            click.echo(f"      {include}")
+
+        click.echo(f"    Files:")
+        for dirpath, _, filenames in os.walk(str(source_root)):
+            for filename in filenames:
+                if filename.startswith('_'):
+                    continue
+                if not filename.endswith(('.scss', '.sass')):
+                    continue
+                source = Path(dirpath) / filename
+                target = (target_root / source.relative_to(source_root)).with_suffix('.css')
+                click.echo(f"      {source} -> {target}")
+                if not dry:
+                    # Import _sass late so that this script can be dry-run without installing
+                    # libsass, which takes a while as it must be compiled from its C source.
+                    from _sass import compile_filename  # pylint: disable=protected-access
+                    success, output, _ = compile_filename(
+                        str(source).encode(fs_encoding),
+                        output_style,
+                        source_comments,
+                        include_paths,
+                        precision,
+                        source_map_filename,
+                        custom_functions,
+                        importers,
+                    )
+                    output_text = output.decode('utf-8')
+                    if not success:
+                        raise Exception(f"Failed to compile {source}: {output_text}")
+                    target.parent.mkdir(parents=True, exist_ok=True)
+                    with open(target, 'w', encoding="utf-8") as target_file:
+                        target_file.write(output_text)
+
+        click.secho(f"    Done.", fg="green")
         # For Sass files without explicit RTL versions, generate
         # an RTL version of the CSS using the rtlcss library.
         for sass_path in glob.glob(str(source) + "/**/*.scss"):
```

Comment:
Yup, the issue is on this line, I should have changed `source` to `source_root`. This made it through my local testing because `source` is a local variable in the new loop above this one, which was erroneously being used here, leading a silent failure.

I imagine that 2U must have a theme without any matched scss files, meaning that the local `source` variable is never assigned, leading a to a loud failure instead in their pipelines instead of a quiet one.